### DISCUSSION TITLE
什么时候能路由导入自定义规则url啊

### DISCUSSION THREAD

**Discussion by rkdawn** [2dust/v2rayNG#3079] [General]
这个位置什么时候能自动导入就好了
![Screenshot_2024-05-04-12-10-09-620_com.v2ray.ang.jpg](https://github.com/2dust/v2rayNG/assets/42829652/7fb92a57-3050-4c62-9b95-d4cd96dc3aed)

**Comment by rkdawn** [comment]
才发现，原来可以自定义了。。在路由设置那![Screenshot_2024-05-04-12-18-01-802_com.v2ray.ang.jpg](https://github.com/2dust/v2rayNG/assets/42829652/c79cdc5d-a85b-49a1-91c4-1e7fd906be47)

**Comment by zazitufu** [comment]
这里导入的URL规则是什么格式的？能给出url里的规则的示例我学习一下吗？谢谢

**Reply by 2dust** [reply]
标准规则，用逗号隔开

**Reply by zazitufu** [reply]
抱歉，我比较愚钝，请问在GEO资源文件里，添加完一个路由规则的url链接，并下载成功了，它是自动就被应用吗？

因为我在路由设置界面看不到有设置使用我下载的成功的路由规则的地方。

![Screenshot_20240517_150650_com v2ray ang](https://github.com/2dust/v2rayNG/assets/57139031/fa626ab7-976a-432e-9770-17925f6a098d)
![Screenshot_20240517_150735_com v2ray ang](https://github.com/2dust/v2rayNG/assets/57139031/dc82952d-c2f2-47ad-b8b4-c8811ecb6dc2)

**Reply by 2dust** [reply]
https://www.v2fly.org/config/routing.html#ruleobject
geosite.dat和geoip.dat中的按上面的规则写

从文件中加载域名：形如 ext:file:tag，必须以 ext: 开头，后面跟文件名和标签，文件存放在[资源目录](https://www.v2fly.org/config/env.html#%E8%B5%84%E6%BA%90%E6%96%87%E4%BB%B6%E8%B7%AF%E5%BE%84)中，文件格式与 geosite.dat 相同，标签必须在文件中存在。

**Reply by zazitufu** [reply]
额，可能是我没有表达清楚，并不是想请教geosite.dat和geoip.dat里面的如何写规则，抱歉。

我是想请教，我在v2rayN的路由里面，点击右键导出的规则，如何在v2rayNG里使用。

我把导出的规则，放在类似https://a.abc.com/myrouting 这里，然后想在v2rayNG里面通过url导入这个规则并使用它。
例如我在v2rayN里面导出的规则如下，保存到上一行的URL文件里：
```
[
  {
    "port": "",
    "outboundTag": "direct",
    "ip": [],
    "domain": [
      "domain:example-example.com",
      "domain:ksjhaoka.com",
      "domain:lot-ml.com",
      "domain:hello.com",
      "domain:idsdz.com",
      "domain:jdbbs.com",
      "domain:vmall.com",
      "domain:bendibao.com",
      "domain:example-example2.com",
      "domain:wyr.me",
      "forcecreat.com",
      "chacuo.net"
    ]
  },
  {
    "port": "",
    "outboundTag": "block",
    "ip": [],
    "domain": [
      "geosite:category-ads-all",
      "gvt2.com"
    ]
  },
  {
    "outboundTag": "direct",
    "domain": [
      "geosite:cn",
      "geosite:geolocation-cn"
    ]
  },
  {
    "outboundTag": "direct",
    "ip": [
      "geoip:private",
      "geoip:cn"
    ]
  },
  {
    "port": "0-65535",
    "outboundTag": "proxy"
  }
]
```

**Comment by 2dust** [comment]
NG的路由没有N那么复杂，参考这个
https://github.com/2dust/v2rayNG/issues/3125

设置中预定义规则选择全局代理或绕过大陆后代理
自定义中增加你的特定domain到直连中

**Reply by zazitufu** [reply]
好的谢谢。主要是我感觉既然N 和 NG都是一家，其实如果这些设置可以互相导出导入，或者可以设置好 一次，两边都可以同时用一套url来导入，那应该更加方便一些。如果作者有空余时间，建议考虑一下这个。
